### PR TITLE
OIDC Integration

### PR DESCRIPTION
**Pull request by sughics**
- Updates to use Java 11
 - OIDC Integration and Grails 4.1.1
 - Update/Migrate usage of older/deprecated functionalities of  Auth and userdetail services. 
 - OpenAPI specs definitions for webservices APIs

### CODE REVIEW

**Comment by sughics** [comment]
Please not that PR also includes the removal of the url mapping `/ws/checkEmailExists`, the function being called i.e. [checkEmailExists()](https://github.com/AtlasOfLivingAustralia/specieslist-webapp/blob/develop/grails-app/controllers/au/org/ala/specieslist/WebServiceController.groovy#L443) and its usage due to security concerns relating to the content returned.  Please contact me for further info this.

The functionality provided to the fron-end by `/ws/checkEmailExists` are now achieved server-side.

File: grails-app/controllers/au/org/ala/specieslist/WebServiceController.groovy [hunk 1]
```diff
@@ -34,21 +50,129 @@ class WebServiceController {
 
     def index() { }
 
+    @Operation(
```
Comment (nickdos) [inline grails-app/controllers/au/org/ala/specieslist/WebServiceController.groovy:53]: 8 space indentation has crept in - ALA coding std is 4 space indentation.

File: grails-app/i18n/messages_de.properties [hunk 2]
```diff
@@ -246,6 +246,7 @@ upload.lists.reload.header=Eine existierende Liste neu laden
 upload.lists.checkdata.errormessage=Ungültiger Dateityp: nur TAB- bzw. Komma-getrennte Textdateien sind erlaubt!
 
 permissions.lists.text=In diesem Formular kann der/die AutorIn der Liste (oder AdministratorIn) weitere BenutzerInnen als <strong>EditorInnen</strong> dieser Liste hinzufügen bzw. löschen. Diese <strong>EditorInnen</strong> dürfen danach ebenfalls die Einträge in der Liste ändern bzw. löschen.
+permissions.lists.warning=Bitte beachten Sie, dass die eingegebene E-Mail-Adresse, wenn sie nicht mit einem vorhandenen ALA-Benutzer �bereinstimmt, beim Speichern nicht zur Berechtigungsliste hinzugef�gt wird.
```
Comment (nickdos) [inline grails-app/i18n/messages_de.properties:249]: You don't need to do non-EN translations - the LA team does this using external i18n files and Vicente monitors commits and adds them to a separate repo.

**Review by nickdos — CHANGES_REQUESTED** [review]
Just a reformat on that one class is needed to fix  indentation

File: grails-app/controllers/au/org/ala/specieslist/WebServiceController.groovy (hunk 1 above)
Comment (sughics) [reply grails-app/controllers/au/org/ala/specieslist/WebServiceController.groovy:53]: Updated

**Review by nickdos — APPROVED** [review]

**Pull request merged**